src/index.d.ts

```diff
@@ -0,0 +1,75 @@
+import React from 'react';
+import * as monacoEditor from 'monaco-editor';
+
+declare class MonacoEditor extends React.Component<MonacoEditorProps> {
+  constructor(props: Partial<MonacoEditorProps>);
+
+  componentDidMount(): void;
+  componentDidUpdate(prevProps: Partial<MonacoEditorProps>): void;
+  componentWillUnmount(): void;
```

Comment:
Since we extend `React.Component`, these lifecycles methods and `render` method should be omitted.

Reply:
:+1: